crates/net/network/src/transactions.rs

```diff
@@ -503,18 +511,20 @@ where
             hashes.truncate(GET_POOLED_TRANSACTION_SOFT_LIMIT_NUM_HASHES);
 
             // request the missing transactions
-            let (response, rx) = oneshot::channel();
-            let req = PeerRequest::GetPooledTransactions {
-                request: GetPooledTransactions(hashes),
-                response,
-            };
+            for hash in hashes {
+                let (response, rx) = oneshot::channel();
+                let req = PeerRequest::GetPooledTransactions {
+                    request: GetPooledTransactions(vec![hash]), // request for a single hash
+                    response,
+                };
 
-            if peer.request_tx.try_send(req).is_ok() {
-                self.inflight_requests.push(GetPooledTxRequestFut::new(peer_id, rx))
-            } else {
-                // peer channel is saturated, drop the request
-                self.metrics.egress_peer_channel_full.increment(1);
-                return
+                if peer.request_tx.try_send(req).is_ok() {
+                    self.transaction_fetcher
+                        .register_inflight(GetPooledTxRequestFut::new(peer_id, hash, rx));
+                } else {
+                    self.metrics.egress_peer_channel_full.increment(1);
+                    return
+                }
             }
```

Comment:
this is suboptimal because this now sends 1 request per hash.

we still want to send as much as possible, but only those that are not in flight.

so the better design here would be something like:

```rust
impl TransactionFetcher {
  fn request_from(&mut self, hashes, peer: &Peer) {
     // 1. filter out inflight hashes, and register the peer as fallback for all inflight hashes
     // 2. request all missing from peer
     // 3. insert hashes to inflight set
  }
}

```

Reply:
sgtm! i tried to flesh out the method `request_from`, one thing I'm not sure about is the return value - do you think that makes sense?